pioneer/packages/joy-forum/src/ViewThread.tsx

```diff
@@ -154,20 +227,18 @@ function InnerViewThread (props: ViewThreadProps) {
     const loadPosts = async () => {
       if (!nextPostId || totalPostsInThread === 0 || thread.isEmpty) return;
 
-      const newId = (id: number | BN) => api.createType('PostId', id);
-      const apiCalls: Promise<Post>[] = [];
-      let id = newId(1);
+      await refreshPostsInThreadCache(nextPostId, api);
+      const mapPostToThread = getPostsIdsInThreadCache();
+
+      const postIdsInThread = mapPostToThread.get(Number(thread.id.toString())) as number[];
+      const postsInThisThreadCalls: Promise<Post>[] = [];
```

Comment:
This next part can be rewritten as:
```
const postsInThisThread = await Promise.all(postIdsInThread ? postIdsInThread.map((postId) => api.query.forum.postById(postId)) : []) as Post[];
```
Of course, you can separate this up if you want to (probably preferable :)).